core/services/ardupilot_manager/flight_controller_detector/Detector.py

```diff
@@ -18,89 +19,114 @@ def _is_root() -> bool:
         return os.geteuid() == 0
 
     @staticmethod
-    def detect_navigator_r3() -> Tuple[bool, str]:
-        """Check if navigator R3 is connected using the sensors on the I²C BUS
+    def get_navigator_if_connected() -> Optional[FlightController]:
+        """Returns Navigator board if connected.
+        Check for connection using the sensors on the I²C and SPI buses.
 
         Returns:
-            (bool, str): True if a navigator is connected, false otherwise.
-                String is always empty
+            Optional[FlightController]: Return FlightController if connected, None otherwise.
         """
-        try:
-            bus = SMBus(1)
-            ADS1115_address = 0x48
-            bus.read_byte_data(ADS1115_address, 0)
 
-            bus = SMBus(4)
-            PCA9685_address = 0x40
-            bus.read_byte_data(PCA9685_address, 0)
-
-            return (True, "")
-        except Exception as error:
-            logger.info("Navigator R3 not detected on I2C bus.")
-            logger.debug(error)
-            return (False, "")
-
-    @staticmethod
-    def detect_navigator_r4() -> Tuple[bool, str]:
-        """Check if navigator R4 is connected using the sensors on the I²C BUS
-
-        Returns:
-            (bool, str): True if a navigator is connected, false otherwise.
-                String is always empty
-        """
-        try:
-            bus = SMBus(1)
-            ADS1115_address = 0x48
-            bus.read_byte_data(ADS1115_address, 0)
-
-            AK09915_address = 0x0C
-            bus.read_byte_data(AK09915_address, 0)
-
-            BME280_address = 0x76
-            bus.read_byte_data(BME280_address, 0)
-
-            bus = SMBus(4)
-            PCA9685_address = 0x40
-            bus.read_byte_data(PCA9685_address, 0)
-
-            return (True, "")
-        except Exception as error:
-            logger.info("Navigator R4 not detected on I2C bus.")
-            logger.debug(error)
-            return (False, "")
+        def is_navigator_r5_connected() -> bool:
+            try:
+                bus = SMBus(1)
+                ADS1115_address = 0x48
+                bus.read_byte_data(ADS1115_address, 0)
+
+                AK09915_address = 0x0C
+                bus.read_byte_data(AK09915_address, 0)
+
+                BME280_address = 0x76
+                bus.read_byte_data(BME280_address, 0)
+
+                bus = SMBus(4)
+                PCA9685_address = 0x40
+                bus.read_byte_data(PCA9685_address, 0)
+                return True
```

Comment:
It's missing the check for mmc5983 via SPI1, the code is the same as Navigator4

Reply:
Yes. I was just placeholding the code there. Still need to check on this.

Reply:
Deprecated R4